pkg/clustermanager/cluster_manager.go

```diff
@@ -934,19 +939,19 @@ func (c *ClusterManager) removeOldWorkerNodeGroups(ctx context.Context, workload
 }
 
 func (c *ClusterManager) InstallCustomComponents(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster, provider providers.Provider) error {
-	if err := c.clusterClient.installCustomComponents(ctx, clusterSpec, cluster); err != nil {
+	if err := c.eksaComponents.Install(ctx, logger.Get(), cluster, clusterSpec); err != nil {
 		return err
 	}
-	fullLifecycleAPI := features.IsActive(features.FullLifecycleAPI())
-	if fullLifecycleAPI {
-		err := c.clusterClient.SetEksaControllerEnvVar(ctx, features.FullLifecycleAPIEnvVar, strconv.FormatBool(fullLifecycleAPI), cluster.KubeconfigFile)
-		if err != nil {
-			return err
-		}
-	}
+
+	// TODO(g-gaston): should this be moved inside the components installer?
 	return provider.InstallCustomProviderComponents(ctx, cluster.KubeconfigFile)
 }
 
+// Upgrade updates the eksa components in a cluster according to a Spec.
+func (c *ClusterManager) Upgrade(ctx context.Context, cluster *types.Cluster, currentSpec, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
```

Comment:
Should this be called `UpgradeCustomComponents`? I see that you removed `upgrader.go`, so if this is replacing that, can this also have other logic in the future if we have other upgrades we need to do within ClusterManager?

Reply:
It probably should yeah
However, this satisfies an interface defined in the workflows, so I kept the name to avoid extending changes elsewhere.
Ideally this operation should go directly to the installer from the workflow and not through cluster manager, but again, that expanded the changes a bit too far.